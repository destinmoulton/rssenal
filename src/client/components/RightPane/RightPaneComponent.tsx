import * as React from "react";

import TopBarComponent from "../RightPane/TopBarComponent";
import EntriesListContainer from "../../containers/RightPane/EntriesListContainer";

import * as Types from "../../types";

export interface IRightPaneMapState {
    filterTitle: string;
}

export interface IRightPaneMapDispatch {}

type TAll = IRightPaneMapDispatch & IRightPaneMapState;

interface IState {
    sortBy: string;
}

class RightPaneComponent extends React.Component<TAll, IState> {
    state = {
        sortBy: "publish_date:asc"
    };

    _handleChangeSort = (e: any, data: any) => {
        this.setState({
            sortBy: data.value
        });
    };

    render() {
        const { filterTitle } = this.props;
        const { sortBy } = this.state;
        return (
            <div>
                <TopBarComponent
                    title={filterTitle}
                    onChangeSort={this._handleChangeSort}
                    sortBy={sortBy}
                />
                <EntriesListContainer sortBy={sortBy} />
            </div>
        );
    }
}

export default RightPaneComponent;
