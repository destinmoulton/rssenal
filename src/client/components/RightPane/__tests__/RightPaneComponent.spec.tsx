import * as React from "react";
import { shallow } from "enzyme";
declare global {
    namespace NodeJS {
        interface Global {
            Headers: () => void;
        }
    }
}

global.Headers = jest.fn().mockImplementation(() => {
    return { append: () => {} };
});
import RightPaneComponent from "../RightPaneComponent";

describe("<RightPaneComponent />", () => {
    it("renders and matches the snapshot", () => {
        const wrapper = shallow(
            <RightPaneComponent filterTitle={"TEST_FILTER_TITLE"} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
