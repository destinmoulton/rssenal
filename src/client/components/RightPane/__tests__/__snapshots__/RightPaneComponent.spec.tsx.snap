// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightPaneComponent /> renders and matches the snapshot 1`] = `
<div>
  <TopBarComponent
    onChangeSort={[Function]}
    sortBy="publish_date:asc"
    title="TEST_FILTER_TITLE"
  />
  <Connect(EntriesListComponent)
    sortBy="publish_date:asc"
  />
</div>
`;
